### PR TITLE
Suggest re-running command with --trace on fail

### LINKED ISSUE #6359: Suggest re-running command with --trace in limited error message
If the user did not use `--trace` and receives an error message, we should suggest that they re-run the command using that trace switch.

### PR DESCRIPTION
**Pull request by ashmaroli**
Closes #6359 

```
                    ------------------------------------------------
      Jekyll 3.6.0   Please append `--trace` to the `build` command
                     for any additional information or backtrace.
                    ------------------------------------------------
```

### CODE REVIEW

File: lib/jekyll/commands/serve.rb [hunk 1]
```diff
@@ -94,7 +90,7 @@ def start(opts)
           if Jekyll.env == "development"
             config["url"] = default_url(config)
           end
-          [Build, Serve].each { |klass| klass.process(config) }
+          process_with_graceful_fail("serve", config, Build, Serve)
```
Comment (ashmaroli) [inline lib/jekyll/commands/serve.rb]: The first argument expected here is not a `String` but the `cmd` object from `prog.command(:serve) do |cmd|`
This is the reason I added a `friction` label to this PR..

Comment (pathawks) [reply lib/jekyll/commands/serve.rb]: I see. So, we need to pass `cmd` from `init_with_program()` to `start()`?

Comment (ashmaroli) [reply lib/jekyll/commands/serve.rb]: Either that or do away with `start` entirely.. i.e. merge `start` into `init_with..`

Comment (pathawks) [reply lib/jekyll/commands/serve.rb]: I imagine it was added to reduce perceived complexity with regard to Rubocop. `start()` should be a private method, and I don't see any problem passing `cmd` to it.

Comment (ashmaroli) [reply lib/jekyll/commands/serve.rb]: Yeah, we have to decide between _changing method signature_ vs _doing away with unnecessary method_

Comment (pathawks) [reply lib/jekyll/commands/serve.rb]: Oh, I'm not worried about backwards compatibility on this one. I can't imagine anybody else making use of `start()`

If you think it makes more sense to inline it, and Rubocop doesn't complain, I'm :+1::+1:

**Comment by ashmaroli** [comment]
@pathawks Do you think I should move the `Do away with the 'start' method` to a separate PR to keep the diff here simpler?

**Comment by pathawks** [comment]
Nah, we’re just sorting out merge conflicts. Let’s keep it all together 👍🏼

**Comment by ashmaroli** [comment]
@oe would like you to re-review as this branch has changed considerably since your last review..
Thanks.

File: lib/jekyll/command.rb [hunk 2]
```diff
@@ -73,6 +73,29 @@ def add_build_options(cmd)
...
+
+      # Run ::process method in a given set of Jekyll::Command subclasses and suggest
+      # re-running the associated command with --trace switch to obtain any additional
+      # information or backtrace regarding the encountered Exception.
+      #
+      # cmd     - the Jekyll::Command to be handled
+      # options - configuration overrides
+      # klass   - an array of Jekyll::Command subclasses associated with the command
+      #
+      # rubocop: disable RescueException
```
Comment (pathawks) [inline lib/jekyll/command.rb]: I wonder if we want to add a comment explaining that we definitely _do_ want to `rescue Exception` here.

I'm just worried that a year from now I might be trying to clean up the code for a new version of Rubocop and forget that this is intentional and not just a bandage.

**Comment by DirtyF** [comment]
@jekyllbot: merge +minor

**Pull request merged**